packages/babel-traverse/test/path/isPure.js

```diff
@@ -0,0 +1,97 @@
+import { parse } from "@babel/parser";
+
+import _traverse from "../../lib/index.js";
+const traverse = _traverse.default;
+
+function getPath(code) {
+  const ast = parse(code, {
+    plugins: [
+      ["decorators", { version: "2021-12", decoratorsBeforeExport: true }],
+      ["recordAndTuple", { syntaxType: "hash" }],
+      "decoratorAutoAccessors",
+      ["pipelineOperator", { proposal: "hack", topicToken: "%" }],
+    ],
+  });
+  let path;
+  traverse(ast, {
+    Program: function (_path) {
+      path = _path;
+      _path.stop();
+    },
+  });
+  return path;
+}
+
+describe("isPure() returns true", () => {
+  it.each([
+    "class C { [0]() {} }",
+    "class C extends class {} {}",
+    "class C { static accessor x = 1 }",
+    "class C { accessor x = f() }",
+    "class C { #x = f() }",
+    "@dec class C {}; function dec () {}",
+    "class C { static target = new.target }",
+  ])(`NodePath(%p).get("body.0").isPure() should be true`, input => {
+    const path = getPath(input).get("body.0");
+    expect(path.node).toBeTruthy();
```

Comment:
This assertion ensures that the pathRef passed in `.get()` is correct. In fact our previous tests on "class X { get foo() { return 1 } }" always succeed because `path.get(body.0.expression)` is an empty NodePath.